src/GitVersionCore.Tests/IntegrationTests/OtherScenarios.cs

```diff
@@ -94,5 +96,33 @@ public void DoNotBlowUpWhenDevelopAndFeatureBranchPointAtSameCommit()
             fixture.InitializeRepo();
             fixture.AssertFullSemver("1.1.0-alpha.1");
         }
+
+        [TestCase(true, false)]
+        [TestCase(true, true)]
+        [TestCase(false, false)]
+        [TestCase(false, true)]
+        public void HasDirtyFlagIfUncommittedChangesAreInRepo(bool createTempFile, bool stageFile)
```

Comment:
This test is a bit confusing since it tests both dirty and non-dirty with a nested `if` statement. It would be clearer if you split this test in two sharply focused tests with no `if (createTempFile)` statement.

Reply:
Sure will do